server/services/__tests__/prisonerInformation.spec.js

```diff
@@ -0,0 +1,420 @@
+const { PrisonerInformationService } = require('../prisonerInformation');
+const { User } = require('../../auth/user');
+
+describe('PrisonerInformation', () => {
+  let prisonApiRepository = {};
+
+  const user = new User({
+    prisonerId: 'PRISONER_ID',
+    firstName: 'Test',
+    surname: 'User',
+    bookingId: 'BOOKING_ID',
+  });
+
+  const transactions = [
+    {
+      paymentDate: '2021-01-03',
+      postingType: 'CR',
+      penceAmount: 10,
+      currency: 'GBP',
+      balance: 30,
+      entryDescription: 'Received some money',
+      agencyId: 'TST',
+    },
+    {
+      paymentDate: '2021-01-02',
+      postingType: 'CR',
+      penceAmount: 10,
+      currency: 'GBP',
+      balance: 20,
+      entryDescription: 'Received some money',
+      agencyId: 'TST',
+    },
+    {
+      paymentDate: '2021-01-01',
+      postingType: 'CR',
+      penceAmount: 10,
+      currency: 'GBP',
+      balance: 10,
+      entryDescription: 'Received some money',
+      agencyId: 'TST',
+    },
+  ];
+
+  const balances = {
+    spends: 123,
+    cash: 456,
+    savings: 789,
+    currency: 'GBP',
+  };
+
+  const prison = {
+    agencyId: 'TST',
+    description: 'Test (HMP)',
+    longDescription: 'HMP Test',
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    prisonApiRepository = {};
+  });
+
+  describe('getTransactionInformationFor', () => {
+    it('returns transaction data', async () => {
+      const prisonerInformationService = new PrisonerInformationService({
+        prisonApiRepository,
+      });
+
+      prisonApiRepository.getTransactionsFor = jest.fn(() =>
```

Comment:
Cos this is a class you could get jest to mock it out for you. 

```js
const PrisonApiRepository = require('some-path/prisonApiRepository')
jest.mock('some-path/prisonApiRepository')

...

const prisonApiRepository = new PrisonApiRepository(null)
prisonApiRepository.getPrisonDetailsFor.mockResolvedValue(prison)

```



```

Reply:
Arrgh.. frustratingly it doesn't support named exports (something we prefer and use all over the place), without defining our own mock implementation, which is what we already have 😞